### DISCUSSION TITLE
Running `pipenv install atri` results in 'The term 'pipenv' is not recognized as the name of a cmdlet, function, script file, or operable program.' error

### DISCUSSION THREAD

**Discussion by darshitac11** [Atri-Labs/atrilabs-engine#338] [Help: Installation & Start] [answered]
Complete error message:

`PS C:\Users\User\my_app> pipenv install atri
pipenv : The term 'pipenv' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ pipenv install atri
+ ~~~~~~
    + CategoryInfo          : ObjectNotFound: (pipenv:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException`

Author: @ipriyanshi1708 (Slack community)

**Comment by darshitac11** [comment, marked as answer]
| | What to do? | How to do it?|
| --- | --- | --- |
| 1 | Deactivate the virtual environment |  Run `exit` (pipenv) or `conda deactivate` (conda) |
| 2 | Add `pipenv` to the PATH environment variable | Follow the instructions shared [here](https://github.com/Atri-Labs/atrilabs-engine/discussions/586) |
| 3 | Activate the virtual environment | Run `pipenv shell` or `conda activate my_env` | 
| 4 | Install `atri CLI` | Run `pipenv install atri==0.0.78` or `conda create -y -n my_env python` |

Additional resources: https://stackoverflow.com/questions/46041719/windows-reports-error-when-trying-to-install-package-using-pipenv/47676529#47676529

**Reply by Khush838** [reply]
how can i add pipenv to path environment variable

**Reply by darshitac11** [reply]
> how can i add pipenv to path environment variable

@Khush838 - check the instructions shared [here](https://github.com/Atri-Labs/atrilabs-engine/discussions/586).

**Reply by RutujaSantikar** [reply]
I did everything as per the instructions but still getting error while doing **pipenv install atri** is 'pipenv' is not recognized as an internal or external command, operable program or batch file.

**Reply by darshitac11** [reply]
@Ankitsgithubs - can you help @RutujaSantikar here?

**Reply by MaiNehaHu** [reply]
I have tried everything mentioned above but, I'm getting the same thing that 'pipenv' is not considered as external or internal command. please help🙏🏻

**Reply by Nishant2209** [reply]
https://stackoverflow.com/questions/46041719/windows-reports-error-when-trying-to-install-package-using-pipenv/47676529#47676529
this can help i guess!

**Reply by Nishant2209** [reply]
> https://stackoverflow.com/questions/46041719/windows-reports-error-when-trying-to-install-package-using-pipenv/47676529#47676529 this can help i guess!
@darshitac11 ma'am can you please update the above answer and add the instruction given in this link or you can also just add the link in the above answer so that every newcomer can get the clarified solution.

**Reply by darshitac11** [reply]
Thanks @Nishant2209 . I have updated the answer.

**Reply by Nishant2209** [reply]
Welcome ma'am

**Comment by Ankitsgithubs** [comment]
sure ma'am..leave it to me !!!

**Comment by sumavati** [comment]
Hello sir I'm also getting same problem can you help me .

**Reply by Ankitsgithubs** [reply]
please dm me on slack with the Screen shot of terminal

**Reply by Shreyaspk9798** [reply]
how to deactivate the virtural environment

**Reply by Nishant2209** [reply]
https://stackoverflow.com/questions/46041719/windows-reports-error-when-trying-to-install-package-using-pipenv/47676529#47676529
try this @Shreyaspk9798

**Comment by ns7754** [comment]
Hello Sir, after setting the environmental variable path for Python and script  error is not resolved [ERROR:'pipenv' is not recognized as an internal or external command,
operable program or batch file.]
while using this command [pip install --user pipenv] it is showing[ Requirement already satisfied: ]
but still I am getting the above  [ERROR:'pipenv' is not recognized as an internal or external command,
operable program or batch file.]
kindly help

**Reply by tanishqawasthi** [reply]
Yes same for me.
Were you able to resolve it? If yes, can you tell me how to

**Reply by Nishant2209** [reply]
https://stackoverflow.com/questions/46041719/windows-reports-error-when-trying-to-install-package-using-pipenv/47676529#47676529

**Comment by ruapps** [comment]
I have added pipenv to the Path environment variable but still its throwing the same error on 'pipenv install atri', how to resolve it?

**Reply by Ankitsgithubs** [reply]
check for the python version ...it should be between Python >=3.8 & <=3.10.8.

if you have installed correct version and still facing the same error ..i guess you had make some mistakes in environment variable.

**Reply by MaiNehaHu** [reply]
But I have done everything correct  environment variable also.

**Reply by Ankitsgithubs** [reply]
please dm me on slack with the Screen shot of terminal

**Reply by ruapps** [reply]
I installed correct version and added to environment variables as per instructions, should I run cmd as an administrator and then run command to install pipenv install atri?

**Reply by Nishant2209** [reply]
https://stackoverflow.com/questions/46041719/windows-reports-error-when-trying-to-install-package-using-pipenv/47676529#47676529
try this @ruapps

**Reply by ruapps** [reply]
thank you, brother

**Reply by Nishant2209** [reply]
no prblm buddy

**Comment by sumavati** [comment]
Why you people are sending me.

**Comment by Shreyaspk9798** [comment]
> Complete error message:
> 
> `PS C:\Users\User\my_app> pipenv install atri pipenv : The term 'pipenv' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again. At line:1 char:1
> 
> * pipenv install atri
> * ```
>     + CategoryInfo          : ObjectNotFound: (pipenv:String) [], CommandNotFoundException
>     + FullyQualifiedErrorId : CommandNotFoundException`
>   ```
> 
> Author: @ipriyanshi1708 (Slack community)

I am not able to follow your instructions

**Reply by Nishant2209** [reply]
https://stackoverflow.com/questions/46041719/windows-reports-error-when-trying-to-install-package-using-pipenv/47676529#47676529

**Comment by krrishexe** [comment]
just add the given deprecated warning path to your local envoirnment variables path , and restart the terminal. 
it'll work.
